### ISSUE TITLE
[$250] [MEDIUM] [Groups] - Group chat with one user is allowed in confirmation page, but not in contact list

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#39317]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.58-4
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to FAB > Start chat
3. Click Add to group with one user
4. Click Create chat
5. Note that user lands on 1:1 DM
6. Go to FAB > Start chat
7. Select Add to group with the same user in Step 3 and another user
8. Click Next
9. In confirmation page, unselect the other user and click Start group

## Expected Result:
If app allows to create a group with just one user, Step 3 should result in the creation of a new group

## Actual Result:
n Step 3, app does not allow the creation of group chat with one user when clicking Add to group with just one user in contact list.
However, in Step 7, when user selects two users in contact list and unselects one user in confirmation page, app allows to create group chat in one user

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/ba594a0d-175f-4a3c-aa77-c465a3429fd7

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01dea4375c99aa51fe</li>
        <li>Upwork Job ID: 1775972935262842880</li>
        <li>Last Price Increase: 2024-04-04</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @s77rt</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by lanitochka17** [comment]
@puneetlath  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by marcaaron** [comment]
Gonna remove the blocker here as there is not really any bad thing that will happen if the user creates a 1:1 DM. 

I sort of agree that we _could_ give them the option to create a Group Chat with just one user - but it feels like the more likely thing is that they will create a 1:1 with them.

Not too sure we need to do anything here right now. cc @JmillsExpensify to see if he has any thoughts.

**Comment by marcaaron** [comment]
@ShridharGoel I'm not 100% sure if that's what we want, but doesn't look too bad. Will wait to get @JmillsExpensify thoughts on this before continuing here.

@shawnborton, @dubielzyk-expensify, @dannymcclain might want to weigh in on this one as well. 

I think it is implied by pressing on "Add to Group" that we want to make a Group whether it has one or more users.

**Comment by dannymcclain** [comment]
🤔 Hmm yeah, do we need to just decide if we want to let people create a group of 2 (that is distinct from their DM with each other)?

I kinda think you _should_ be able to create a group with just 2 people, I _think_. But I'm curious what everyone else thinks.

**Comment by shawnborton** [comment]
Yeah, I think a Group can be 2 people because at a later date you might invite more people to it.

**Comment by dubielzyk-expensify** [comment]
I actually kinda think we shouldn't. Cause it allows you to create the same thing as a DM. I even think that if you just add one person to a group, we should toss you over to a DM with them.

Messenger does something similar, but forces you to not create a group with 1 person. 

I know it doesn't specifically impact IOUs, but having a DM with Danny and a Group named Danny and Jon, then the IOU goes to my DM with Danny just feels like more confusion. 

I don't feel super strongly here so definitely happy to lean on consensus here.

**Comment by JmillsExpensify** [comment]
Phew, tough one. I mostly think it's ok that you can create a group with one person, which might also be someone that you also have a DM with. That's why the original scope calls for requiring at least one other person. Equally, the group chat avatar is distinct from the DM avatar, so even if you accidentally created a group there are several UI patterns that help you understand that one is not the same as the other. 

All in all, I think the solution is for us to update the Group Chat creation flow, as I think the existing pattern is confusing and already being deprecated for Splits, so we should deprecate it for Groups too, though it's out of scope for this issue.

**Comment by shawnborton** [comment]
Yeah.... I can see both sides but one use case that comes to mind: let's say a friend and I are planning an event, and we first want to make a Group chat and grab the QR code so that anyone who is coming to the event can join. If we require at least 3 people to make a Group, that means we'd need another organizer to jump in before we can make the group and get our shareable QR code for others to jump in. 

So I mean wild 🌶️ take... but could a group be created with even just one person?! Maybe it should be its own item in the global create flow, and the first step is just naming the group (and updating the avatar if you want), and then we ask for group members as an optional piece.

**Comment by dubielzyk-expensify** [comment]
Well, I'm convinced 😄  I think you make a compelling argument there @shawnborton. If we push it at places like conventions then creating a group ahead of time seems like a valid use case.

**Comment by dannymcclain** [comment]
> Yeah.... I can see both sides but one use case that comes to mind: let's say a friend and I are planning an event, and we first want to make a Group chat and grab the QR code so that anyone who is coming to the event can join. If we require at least 3 people to make a Group, that means we'd need another organizer to jump in before we can make the group and get our shareable QR code for others to jump in.
> 
> So I mean wild 🌶️ take... but could a group be created with even just one person?! Maybe it should be its own item in the global create flow, and the first step is just naming the group (and updating the avatar if you want), and then we ask for group members as an optional piece.

@shawnborton my thinking was the _exact_ same. I almost mentioned creating a group with no one else in it but didn't want to stir the pot too hard 🤣

**Comment by dubielzyk-expensify** [comment]
Sounds good to me!

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

enable the "Next" button to show up if 1 or more contacts are selected.

### What is the root cause of that problem?
 according to the discussion conclusion [above](https://github.com/Expensify/App/issues/39317#issuecomment-2035679897) , we need to modify the behavior of showing the "Next" button in the start chat page, where we need to make it appears when 1 or more contacts are selected.


### What changes do you think we should make in order to solve the problem?



we need to modify the `confirmButtonText` here to:
https://github.com/Expensify/App/blob/d4da2d7cadbe1c620c0c4b9c0760ac1775b64b9f/src/pages/NewChatPage.tsx#L297
```js
confirmButtonText={selectedOptions.length > 0 ? translate('common.next') : translate('newChatPage.createChat')}
```

and the `onConfirmSelection` condition to:
https://github.com/Expensify/App/blob/d4da2d7cadbe1c620c0c4b9c0760ac1775b64b9f/src/pages/NewChatPage.tsx#L299

```js
onConfirmSelection={selectedOptions.length > 0 ? navigateToConfirmPage : createChat}
```


this way we allow the user to create a group with 1 or more other contacts.

..........................................................................................................................................................

Extra: to enable creating a group that consist of only one person, we need to modify the `showConfirmButton` in the group confirm page so that it shows up even if there is only one person in the group setup https://github.com/Expensify/App/blob/4ea37b6c8eb7eb7a2ee4a7678face11a69c0d535/src/pages/NewChatConfirmPage.tsx#L135

```js
showConfirmButton={selectedOptions.length > 0}
```

**Comment by marcaaron** [comment]
@s77rt @abzokhattab adding you guys to this one too since we are already working together on  https://github.com/Expensify/App/issues/39560

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by s77rt** [comment]
PR deployed to production 2 weeks ago. This is awaiting payment

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week) Good to close?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fourth week)

**Comment by s77rt** [comment]
This is still due payment. @marcaaron Can you please assign a BZ member?

**Comment by mallenexpensify** [comment]
@s77rt  @abzokhattab , can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~0123a0fa7b4a2fcc25

**Comment by s77rt** [comment]
@mallenexpensify Accepted! Thanks!

**Comment by abzokhattab** [comment]
thank u applied

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fifth week)

**Comment by mallenexpensify** [comment]
Contributor: @abzokhattab paid $250 via Upwork
Contributor+: @s77rt paid $250 via Upwork.

@s77rt plz complete the BZ checklist below? Thx


**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@] Determine if we should create a regression test for this bug.
- [ ] [@] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@BZ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: https://github.com/Expensify/App/pull/37458
- The offending PR has been commented on: https://github.com/Expensify/App/pull/37458#discussion_r1686929796
- A discussion in #expensify-bugs has been started: Not needed (unique bug)
- Determine if we should create a regression test for this bug: No. We already have a test case covering that https://expensify.testrail.io/index.php?/cases/view/2645392

**Comment by s77rt** [comment]
@mallenexpensify Done! ^

**Comment by mallenexpensify** [comment]
Thanks @s77rt , closing
